enzy_htp/structure/structure_region/structure_region.py

```diff
@@ -311,6 +318,65 @@ def topology(self) -> Structure:
         for atom in self.atoms:
             if not atom.parent.is_residue_cap():
                 return atom.root()
+            
+    def convert_to_structure(self, cap_as_residue = False) -> Structure:
+        """Converts all atoms in the region to a Structure
+        Args:
+            cap_as_residue: will not include the cap in the residue if True
+        Returns:
+            The corresponding Structure of the structure region"""
+        residue_mapper: {Residue, List[Atom]} = {}
+        res_to_cap_atoms: {Residue, int} = {}
+
+        for aa in self.atoms:
+            if isinstance(aa.parent, ResidueCap):
+                if cap_as_residue:
+                    continue
+                else:
+                    aa.residue = aa.parent.link_atom.residue
+                if aa.residue not in res_to_cap_atoms:
+                    res_to_cap_atoms[aa.residue] = 0
+                res_to_cap_atoms[aa.residue] += 1
+
+            if aa.residue not in residue_mapper:
+                residue_mapper[aa.residue] = []
+            residue_mapper[aa.residue].append(aa)
+
+        residues = []
+        for res, atoms in residue_mapper.items():
+            if res not in res_to_cap_atoms.keys():
+                res_to_cap_atoms[res] = 0
+            
+            if res.rtype == chem.ResidueType.MODIFIED:
```

Comment:
Please use deepcopy() instead. And change the .atoms to the atoms in the mapper. This way all original attributes are kept.

Reply:
If you made this by mimiking the way we construct in pdb_io, it is a great effort! However, we would like to treat this with fewer lines to make it more readable and extensible (i.e.: we may add new residue types)